lib/rubocop/cop/performance/use_zip_to_wrap_array_contents.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for `.map { |id| [id] }` and suggests replacing it with `.zip`.
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { |id| [id] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { [_1] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| id }
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| [id, id] }
+      class UseZipToWrapArrayContents < Base
+        extend AutoCorrector
+
+        RESTRICT_ON_SEND = %i[map].freeze
+        MSG = 'Use `.zip` instead of `.map { |id| [id] }`.'
+
+        # Matches regular block form `.map { |e| [e] }`
+        # @!method map_with_array?(node)
+        def_node_matcher :map_with_array?, <<~PATTERN
+          (block
+            (send _ :map)
+            (args (arg _id))
+            (array (lvar _id)))
+        PATTERN
+
+        # Matches numblock form `.map { [_1] }`
+        # @!method map_with_array_numblock?(node)
+        def_node_matcher :map_with_array_numblock?, <<~PATTERN
+          (numblock
+            (send _ :map)
+            1
+            (array (lvar _))
+          )
+        PATTERN
+
+        def on_send(node)
+          return unless node.method?(:map)
+
+          parent = node.parent
+          return unless parent
+
+          register_offense(parent) if parent.block_type? && map_with_array?(parent)
+        end
+
+        def on_numblock(node)
+          return unless node.method?(:map)
+          return unless map_with_array_numblock?(node)
+
+          register_offense(node)
+        end
+
```

Comment:
```suggestion
          register_offense(parent) if map_with_array?(parent) || map_with_array_numblock?(parent)
```

I'm sure you can combine the pattern matchers somehow but I'm not good enough with them to tell how

Reply:
Thanks, great catch! I added a spec for this. It also fails with a missing receiver, so I am adding the correct guard clause for that as well.